app/src/main/java/com/codesoom/assignment/TaskManager.java

```diff
@@ -0,0 +1,62 @@
+package com.codesoom.assignment;
+
+
+import com.codesoom.assignment.Exception.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Task 목록 정보를 저장하고 있으며, Task 목록에 대한 CRUD 처리를 담당합니다.
+ */
+public class TaskManager {
+    private Long nextId = 1L;
+    private HashMap<Long, Task> tasks = new HashMap<>();
+
+    private final Object nextIdLock = new Object();
+
+    public Task create(String title) {
+        synchronized (nextIdLock) {
+            nextId++;
+        }
+
+        Long newId = nextId;
+        Task newTask = new Task(newId, title);
+        this.tasks.put(newId, newTask);
+
+        return newTask;
+    }
+
+    public List<Task> getAll() {
+        return new ArrayList<Task>(this.tasks.values());
+    }
+
+    public Task getOne(Long id) throws TaskNotFoundException {
+        if(!exist(id)) {
+            String message = "getOne() 메소드에서 요청하신 id(" + id + ")에 대한 Task에 접근할 수 없습니다.";
```

Comment:
호출 메소드 이름을 에러 메시지에 입력하는 것은 지나칩니다. 이렇게 하면 메소드 이름과 에러 메시지가 커플링이 생기죠. 메시지는 에러가 발생한 장소가 아니라 에러가 발생한 원인이 된 데이터에 집중하는 것이 좋습니다. 게다가 스택 트레이스를 출력하게 되면 어느 메소드에서 발생했는지는 전부 출력되므로 굳이 메소드 이름을 적을 필요도 없습니다. 그리고 이 메소드가 예외를 `throw` 던지고 있다는 사실을 떠올려봅시다. 이 예외를 누군가가 `catch`하겠죠? 그런데 예외를 받는 곳에서 이 예외를 번역해서 또다른 예외로 던지게 된다면 어떻게 될까요?

```suggestion
            String message = "id(" + id + ")에 대한 Task에 접근할 수 없습니다.";
```

Reply:
저번에 너무 성의 없게 메세지를 작성하다가 이번에는 자세하게 메세지를 작성해보자 라는 생각으로 과도하게 작성해 버렸네요.

Reply:
연속으로 throw를 던지게 되면 해당 문구가 의미가 없겠네요.